packages/experience/src/hooks/use-single-sign-on-watch.ts

```diff
@@ -1,31 +1,30 @@
 import { SignInIdentifier, type SsoConnectorMetadata } from '@logto/schemas';
-import { useEffect, useState, useCallback, useContext } from 'react';
-import { type Control, useWatch } from 'react-hook-form';
+import { useEffect, useCallback, useContext } from 'react';
 import { useNavigate } from 'react-router-dom';
 
 import SingleSignOnContext from '@/Providers/SingleSignOnContextProvider/SingleSignOnContext';
+import SingleSignOnFormModeContext from '@/Providers/SingleSignOnFormModeContextProvider/SingleSignOnFormModeContext';
 import { getSingleSignOnConnectors } from '@/apis/single-sign-on';
+import type { IdentifierInputValue } from '@/components/InputFields/SmartInputField';
 import { singleSignOnPath } from '@/constants/env';
 import useApi from '@/hooks/use-api';
 import useSingleSignOn from '@/hooks/use-single-sign-on';
 import { validateEmail } from '@/utils/form';
 
-import type { FormState } from './index';
+import { useSieMethods } from './use-sie';
 
-const useSingleSignOnWatch = (control: Control<FormState>) => {
+const useSingleSignOnWatch = (identifierInput?: IdentifierInputValue) => {
   const navigate = useNavigate();
+
+  const { singleSignOnEnabled } = useSieMethods();
+
   const { setEmail, setSsoConnectors, ssoConnectors, availableSsoConnectorsMap } =
     useContext(SingleSignOnContext);
-  const [showSingleSignOn, setShowSingleSignOn] = useState(false);
-  const request = useApi(getSingleSignOnConnectors);
-  const singleSignOn = useSingleSignOn();
 
-  const isSsoEnabled = availableSsoConnectorsMap.size > 0;
+  const { showSingleSignOnForm, setShowSingleSignOnForm } = useContext(SingleSignOnFormModeContext);
```

Comment:
replace the original useState with the useContext. As we need to share the `showSingleSignOnForm` status with upper scope.